openhands/server/routes/github.py

```diff
@@ -0,0 +1,61 @@
+import requests
+from fastapi import APIRouter, HTTPException, Request
+from fastapi.responses import JSONResponse
+
+from openhands.server.shared import openhands_config
+from openhands.server.types import AppMode
+
+app = APIRouter(prefix='/api')
+
+
+@app.get('/github/repositories')
+def get_github_repositories(
+    request: Request,
+    page: int = 1,
+    per_page: int = 10,
+    sort: str = 'pushed',
+    installation_id: int | None = None,
+):
+    # Extract the GitHub token from the headers
+    github_token = request.headers.get('X-GitHub-Token')
+    if not github_token:
+        raise HTTPException(status_code=400, detail='Missing X-GitHub-Token header')
+
+    if installation_id is None and openhands_config.APP_MODE == AppMode.SAAS:
```

Comment:
A bit nitpicky, but could we do this check like the `verify()` method does for path, in the new config class? It is a sort of sanity verification after all